src/items.py

```diff
@@ -1,15 +1,30 @@
 import logging
 
 from typing import List
-from collections import namedtuple
+from copy import deepcopy
+
 
 log = logging.getLogger()
 
 
 # Structure for item: Each item has a name and id
-Item = namedtuple('Item', ['name', 'id'])
+class Item:
+    def __init__(self, name, id):
+        self.name = name
```

Comment:
can we please use type. name can be misleading. i.e. name might be misunderstood as unique to each item whereas it's really the type of the item and id is the unique identifier.